src/qt/coincontroldialog.cpp

```diff
@@ -225,10 +225,12 @@ void CoinControlDialog::setModel(WalletModel* model)
 // (un)select all
 void CoinControlDialog::buttonSelectAllClicked()
 {
-    Qt::CheckState state = Qt::Checked;
+    // "Select all": if some entry is unchecked, then check it
+    // "Unselect all": if some entry is checked, then uncheck it
+    Qt::CheckState state = ui->pushButtonSelectAll->text() == tr("Select all") ? Qt::Unchecked : Qt::Checked;
```

Comment:
would be good to use a boolean here, hardcoded texts are not good for statements.